### PR TITLE
feat: add /waypoint/api-docs

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

This PR stubs in content at [/waypoint/api-docs][/waypoint/api-docs], pulling from [the Waypoint server swagger file][swagger].

[task]: https://app.asana.com/0/1202097197789424/1203976641189855/f
[swagger]: https://github.com/hashicorp/waypoint/blob/main/pkg/server/gen/server.swagger.json
[preview]: https://dev-portal-git-zswaypoint-api-docs-hashicorp.vercel.app/
[/waypoint/api-docs]: https://dev-portal-git-zswaypoint-api-docs-hashicorp.vercel.app/waypoint/api-docs

### CODE REVIEW

**Review by demophoon — APPROVED** [review]
The Waypoint swagger is auto generated from our protos under a single service so the formatting is expected with regard to the sidebar as opposed to other services like Boundary

**Review by kylemac — APPROVED** [review]
obligatory 👍 . britt's was the meaningful one

File: src/components/open-api-page/utils/remove-circular-references.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * We should dig into why this is needed, and figure out where it might
+ * be breaking or compromising the API docs UI, which seems possible.
+ *
+ * Alternately, we should refactor the OpenAPI docs page to be more explicit
+ * in the data types it requires, and do more of the schema processing
+ * server-side in `getStaticProps`. There should be no need to pass
+ * circular references as static props for rendering (as that would result
+ * in an infinitely deep properties being rendered, for example), and we
+ * should likely address and resolve that intentionally.
+ */
```
Comment (thiskevinwang) [inline src/components/open-api-page/utils/remove-circular-references.js:38]: 💭 It seems really strange that we have to handle this.

@zchsh wanna create an Asana task for this?

Comment (zchsh) [reply src/components/open-api-page/utils/remove-circular-references.js:38]: Great call - I've made 🎟 [[api-docs] investigate circular references in Waypoint API docs](https://app.asana.com/0/1202097197789424/1203989531295664/f)

**Pull request merged**